### DISCUSSION TITLE
Does Velox support custom file systems?

### DISCUSSION THREAD

**Discussion by rshkv** [facebookincubator/velox#6175] [Q&A] [answered]
We have a custom file system scheme (`custom://`) for which, in Java/Spark, we implement a custom Hadoop file system client.

I was wondering if we could connect to our custom file system from Velox. (In the context of Gluten.)

It seems we could implement the `FileSystems.h` interface. But I'm unsure if `FileSystem's` `registerFileSystem` is meant to be used outside of Velox.

**Comment by rshkv** [comment]
I learned that the Gluten folks already [do call `registerFileSystem`](https://github.com/oap-project/gluten/blob/cf0a50616ab0b84e336c4ec8ebdab1686102f1e6/cpp/velox/jni/JniFileSystem.cc#L465) to register their own file system. Still curious if from Velox' point of view that's meant to be public API.

(For other folks stumbling over this trying to make Velox work with Gluten, I asked about custom file systems in Gluten: https://github.com/oap-project/gluten/discussions/2832.)

**Reply by majetideepak** [reply, marked as answer]
Yes, you can register your own filesystem. All Velox registration APIs are meant to be public afaik.
You can see some more examples here https://github.com/facebookincubator/velox/tree/main/velox/connectors/hive/storage_adapters
